pages/mesosphere/dcos/services/nifi/1.0.0-1.9.2/security/index.md

```diff
@@ -0,0 +1,141 @@
+menuWeight: 49
+excerpt: DC/OS NiFi Service encryption, authentication, and authorization
+featureMaturity:
+enterprise: false
+model: /mesosphere/dcos/services/nifi/data.yml
+render: mustache
+---
+
+# DC/OS {{model.techName }} Security
+
+The DC/OS {{model.techName }} service supports DC/OS {{model.techName }}’s native transport encryption, authentication, and authorization mechanisms. The service provides automation and orchestration to simplify the usage of these important features.
+
+A good overview of these features can be found  [here](https://{{ model.serviceName }}.apache.org/docs/{{ model.serviceName }}-docs/html/administration-guide.html).
+
+## Transport Encryption and Kerberos Authentication
+With transport encryption enabled, DC/OS {{model.techName }} will automatically deploy all nodes with the correct configuration to encrypt communication via SSL. The nodes will communicate securely between themselves using SSL. SSL authentication requires that all DC/OS {{model.techName }} Nodes present a valid certificate from which their identity can be derived for communicating between themselves.
+DC/OS {{model.techName }} uses the CN of the SSL certificate as the principal for a given Node.
+For example, CN={{ model.serviceName }}-0-node.demonifi, O="Mesosphere, Inc", L=San Francisco, ST=CA, C=US.
+
+The service uses the [DC/OS CA](/mesosphere/dcos/latest/security/ent/tls-ssl/) to generate the SSL artifacts that it uses to secure the service. Any client that trusts the DC/OS CA will consider the service’s certificates valid.
+
+<p class="message--note"><strong>NOTE: </strong>Enabling transport encryption is required to use <a href="/mesosphere/dcos/latest/security/ent/tls-ssl/">SSL authentication</a> for authentication, but is optional for Kerberos authentication.</p>
+
+## Prerequisites
+- A DC/OS Service Account with a secret stored in the [DC/OS Secret Store.](/mesosphere/dcos/latest/security/ent/service-auth/custom-service-auth/)
+- DC/OS superuser permissions for modifying the permissions of the Service Account.
+
+## Configure Transport Encryption
+
+<p class="message--note"><strong>NOTE: </strong>A complete guide to Configuring DC/OS Access for DC/OS {{model.techName }} can be found below.</p>
+
+
+## Set up the service account
+
+[Grant](/mesosphere/dcos/latest/security/ent/perms-management/) the service account the correct permissions.
+- In DC/OS 1.10, the required permission is `dcos:superuser full`.
+- In DC/OS 1.11 and later, the required permissions are:
+```shell
+dcos:secrets:default:/<service name>/* full
+dcos:secrets:list:default:/<service name> read
+dcos:adminrouter:ops:ca:rw full
+dcos:adminrouter:ops:ca:ro full
+```
+where `<service name>` is the name of the service to be installed.
+
+Run the following DC/OS Enterprise CLI commands to set permissions for the service account on a strict cluster:
+
+```shell
+dcos security org users grant ${SERVICE_ACCOUNT} dcos:mesos:master:task:app_id:<service/name> create
```

Comment:
I think we need a command to cover the case which caused https://jira.mesosphere.com/browse/DCOS-60595
According to https://github.com/mesosphere/dcos-commons/releases/tag/0.57.0 something like this will be needed:
```
dcos security org users grant ${SERVICE_ACCOUNT} dcos:mesos:master:reservation:role:${FOLDER_NAME} create
```

Reply:
To clarify, on DC/OS 2.0 the set of necessary roles is *different* than on previous DC/OS versions.
It goes like this:
```
    dcos security org users grant ${SERVICE_ACCOUNT} dcos:mesos:agent:task:user:${USER} create
    dcos security org users grant ${SERVICE_ACCOUNT} dcos:mesos:master:task:user:${USER} create
    dcos security org users grant ${SERVICE_ACCOUNT} dcos:mesos:master:framework:role:${FOLDER_ROLE} create
    dcos security org users grant ${SERVICE_ACCOUNT} dcos:mesos:master:reservation:principal:${SERVICE_ACCOUNT} delete
    dcos security org users grant ${SERVICE_ACCOUNT} dcos:mesos:master:reservation:role:${FOLDER_ROLE} create
    dcos security org users grant ${SERVICE_ACCOUNT} dcos:mesos:master:volume:principal:${SERVICE_ACCOUNT} delete
    dcos security org users grant ${SERVICE_ACCOUNT} dcos:mesos:master:volume:role:${FOLDER_ROLE} create
```

In the 2.0 case, `FOLDER_ROLE` is the name of the top-level group where the service is located. In previous versions it's `fully__qualified__path__service-role`.

Reply:
So is this issue resolved?